### PR TITLE
Cleanup so that we can use it

### PR DESCRIPTION
**Pull request by ukstv**
- Remove unnecessary dependencies, especially ts-jest from production dependencies
- Remove generated dist and lib folders from Git version control
- Respect linter

Note: it is still a CJS module. It would be nice if you can make it ES module.

### CODE REVIEW

**Comment by bshambaugh** [comment]
Thanks for looking into this @ukstv . I'll use this as motivation to crack open this code-base for a second look in preference of the motive of  having quality code for your libraries.

Here are some unfilterered thoughts (sans moving the 2 most relevant blurbs to the top of the stack).

I am not an expert with tsconfig.json and package.json files. I literally merged parts of other configuration files that seemed to work not knowing what every configuration option meant.

I know this code worked in the js-ceramic environment without complaining from the linter or test framework. In frustration, and with a deadline I built this technical debt.

Ceramic has very well funded teams of skilled developers building with it as well as countless people experimenting with it at hackathons. Forgive me if I have had a mindset that my work is an insignificant part of this ocean.
It may in fact be that my endeavors with ESP32 devices tying into Ceramic play an irrelvant role in the future of Ceramic and web3, but I know it is important that libraries used by all are of high quality. 

Thanks for looking into this at this level and keeping me honest.

This sloppy code was result of me banging on this code until it worked with a pressing deadline of November 14th, which is the fall Internet Identity Workshop (IIW) with a demonstration.

Thanks for not validating the pejorative mindset I had been holding where I hold the only pair of eyes that would be willing to look at this codebase.

First things first with the linter, tsconfig.json, and package.json. Later I may pepper you (or others in the ceramic community) with questions so I have at least a decent demonstration of an
ESP32 based cryptographic signer working with ceramic over websockets at IIW.

**Comment by ukstv** [comment]
Oh, that's a lot of words. Your contribution is very valuable to me personally, and to Ceramic community at large.

This library is quite well written, do not be harsh on yourself. The changes I made are really small nitpicks. Even senior developers struggle with all the settings available considering pace of change in the ecosystem.

It would be nice if you could merge and release the changes, as they do not change the logic. And then some time after you are done with your project, that ESM suggestion I made could be taken into consideration. I am really looking forward seeing your project a success.

**Comment by bshambaugh** [comment]
The linter still gives warnings for any, but it may be minor. 

line 28 in nist_weierstrauss_common.test.ts looks like this:
expect(publicKey_u8a).toEqual(true);

will not fix for now.

/home/ubuntu/Downloads/nov22nd/nist-weierstrauss/src/__tests__/nist_weierstrauss_common.test.ts
   11:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   32:31  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   38:31  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   51:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   58:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   65:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   85:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   98:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  105:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  112:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  119:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  126:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  145:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  152:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  216:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  229:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  236:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  243:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  250:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  263:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  270:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/ubuntu/Downloads/nov22nd/nist-weierstrauss/src/__tests__/secp256r1.test.ts
   6:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  19:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/ubuntu/Downloads/nov22nd/nist-weierstrauss/src/__tests__/secp384r1.test.ts
   6:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  19:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/ubuntu/Downloads/nov22nd/nist-weierstrauss/src/__tests__/secp521r1.test.ts
  18:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

**Comment by ukstv** [comment]
The warnings are kind of okay. I had a look into them.

**Comment by bshambaugh** [comment]
Well truth be told @ukstv , I was looking at this last night before hitting the bed and I had no idea how specify the types for the test files. It was trying to follow stuff like: https://www.typescriptlang.org/docs/handbook/basic-types.html but to no avail. Then I told myself it was in a test file, not in the body of the code, and just a warning. To get something out last night I just merged so there would be something for you and told myself I would worry about it tomorrow (today now). 

tl;dr:
But if it is kind of okay... then maybe abandon perfection for now? Always willing to fix this.

**Comment by ukstv** [comment]
> tl;dr: But if it is kind of okay... then maybe abandon perfection for now? Always willing to fix this.

That definitely goes to "Someday Maybe" bucket.
